### PR TITLE
Adding log-out button

### PR DESCRIPTION
**Pull request by juliajuju93**
Added log-out button

- added signOut function (async function) that trigger the sign-out process using authService.
-> It updates the session and the isAuthenticated state to reflect the unauthenticated status. Additionally, it performs client-side cleanup by removing the API endpoint, client ID, and tenant ID from local storage. Finally, it refreshes the URL to the original state

### CODE REVIEW

File: src/components/Header/index.tsx [hunk 1]
```diff
@@ -38,11 +33,24 @@ const Header = () => {
     const signIn = async () => {
         await authService.signIn();
         session.setAuthenticated(true);
+        setIsAuthenticated(true);
+    };
+
+    const signOut = async () => {
+        await authService.signOut();
+        session.setAuthenticated(false);
+        setIsAuthenticated(false);
```
Comment (malincrist) [inline src/components/Header/index.tsx:42]: not directly related to this PR, but I'm curious if there is a specific reason we need both the global state (via `useSession()`) and the local `useState` in this component. Is there a case where the local state provides a benefit that the global state does not? If not, would it make sense to refactor to use only `useSession()` for better consistency and simplicity?

**Review by malincrist — APPROVED** [review]

File: src/components/Header/index.tsx [hunk 2]
```diff
@@ -38,11 +33,24 @@ const Header = () => {
...
+        // Client-side cleanup
+        dataApiEndpoint.remove();
+        dataApiClientId.remove();
+        dataApiTenantId.remove();
```
Comment (malincrist) [inline src/components/Header/index.tsx:46]: why do we need to clean-up these?

File: src/components/Header/index.tsx (hunk 1 above)
Comment (jsorohova) [reply src/components/Header/index.tsx:42]: I won't be able to answer right now why but the state is using authService value, not the session one. So I would leave it as is for now and later think if we can improve it.

File: src/components/Header/index.tsx (hunk 2 above)
Comment (jsorohova) [reply src/components/Header/index.tsx:46]: To remove keys from local storage. This is a good thing to do.

File: src/components/Header/index.tsx [hunk 3]
```diff
@@ -1,8 +1,3 @@
-/*---------------------------------------------------------------------------------------------
- *  Copyright (c) Microsoft Corporation. All rights reserved.
- *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
- *--------------------------------------------------------------------------------------------*/
-
```
Comment (jsorohova) [inline src/components/Header/index.tsx:5]: Do we need to remove it? I believe Pierce added this (not completely sure though who did). This is in every file.

**Review by jsorohova — APPROVED** [review]

File: src/components/Header/index.tsx (hunk 2 above)
Comment (jsorohova) [reply src/components/Header/index.tsx:46]: Good point. Then yes, need to understand if we need to remove the keys upon sign out. Probably not.

File: src/components/Header/index.tsx (hunk 3 above)
Comment (juliajuju93) [reply src/components/Header/index.tsx:5]: I think we can leave it for now!

**Review by jsorohova — APPROVED** [review]

**Pull request merged**